API/CopilotService/__init__.py

```diff
@@ -40,56 +42,66 @@ def main(req: func.HttpRequest, context: func.Context) -> func.HttpResponse:
         history = get_param(req, 'history', default=[])
         top_num = get_param_int(req, 'top_num', default=5)
         service_type = get_param_enum(req, 'type', ServiceType, default=os.environ.get("DEFAULT_SERVICE_TYPE", ServiceType.GPT_GENERATION))
-    except ParameterException as e:
-        logger.error(f'Response Status 400: ParameterException: {e.msg}')
-        return func.HttpResponse(e.msg, status_code=400)
+
+        result = copilot_service(context, question, history, top_num, service_type)
+    except RequestException as e:
+        logger.error(f'Error: UserException: {e.msg}', exc_info=e)
+        return func.HttpResponse(e.to_response_body(), status_code=400)
+    except CopilotException as e:
+        logger.error(f'Error: ServiceException: {e.msg}', exc_info=e)
+        return func.HttpResponse(e.to_response_body(), status_code=500)
```

Comment:
```suggestion
    except ServiceException as e:
        logger.error(f'Error: ServiceException: {e.msg}', exc_info=e)
        return func.HttpResponse(e.to_response_body(), status_code=500)
```

Reply:
The `CopilotException` contains wider Exception than its child `ServiceException`. We could respond with detailed info when the service raise a raw `CopilotException` or newly defined `Exception` based on `CopilotException`.
I think both approaches are reasonable.